src/providers/v2v3/V2V3ContractsProvider.tsx

```diff
@@ -1,14 +1,26 @@
-import { CV_V2 } from 'constants/cv'
 import { V2V3ContractsContext } from 'contexts/v2v3/V2V3ContractsContext'
 import { useV2V3ContractLoader } from 'hooks/v2v3/V2V3ContractLoader'
+import { V2CVType, V3CVType } from 'models/cv'
+import { useState } from 'react'
 
-export const V2V3ContractsProvider: React.FC = ({ children }) => {
-  const contracts = useV2V3ContractLoader({ cv: CV_V2 })
+export const V2V3ContractsProvider: React.FC<{
+  initialCv: V3CVType | V2CVType
+}> = ({ children, initialCv }) => {
+  const [cv, setCv] = useState<V2CVType | V3CVType>(initialCv)
+
+  const contracts = useV2V3ContractLoader({ cv })
```

Comment:
The idea here is that, when the `cv` changes, we load the new contracts for that cv. Users can switch the version using this component: [ContractVersionSelect.tsx](https://github.com/jbx-protocol/juice-interface/pull/1995/files#diff-5c5f61fc15af2d631c111baab4fdde05fdc57d93a11c6c97b5dba09496bc9a5f).

For example, if we're on V2, and then the user selects V3, we would reload the contracts, which would trigger a re-render of the whole component tree and render the project with the V3 data.

Reply:
Ok sweet, got it